### PR TITLE
COMPAT locale provider will be removed in a future release

### LINKED ISSUE #11550: COMPAT locale provider will be removed in a future release
From https://bugs.openjdk.org/browse/JDK-8305402

> COMPAT locale provider has remained to keep the backward compatibility of JDK 8's locale data, since CLDR-based locale data became the default in JDK 9. The intention was to encourage applications to migrate to CLDR, so for future removal of the COMPAT locale provider, acknowledging the user of future removal is recommended.

**To Reproduce**
1. Run tests on a plugin using JDK 21 
2. Observe multiple warnings:
```
> Task :test
Dec 09, 2023 12:30:31 PM sun.util.locale.provider.LocaleProviderAdapter <clinit>
WARNING: COMPAT locale provider will be removed in a future release
```

**Expected behavior**
No warnings.

**Plugins**

Error occurs when testing any plugin which inherits from `OpenSearchTestBasePlugin`

**Host/Environment (please complete the following information):**
 - OS: macOS 14.1.2 (aarch64)
 - JDK Version 21.0.1

**Additional context**

The locale is set here:

https://github.com/opensearch-project/OpenSearch/blob/3c14cc187525fde6cc24fd5e3acec18c44ce9a1d/buildSrc/src/main/java/org/opensearch/gradle/OpenSearchTestBasePlugin.java#L113

The fix is to simply replace `COMPAT` with `CLDR`.   _Some tests will break and will need to be fixed._

Note that the "breaking change" here was the JDK default behavior, so this only breaks our tests which explicitly set COMPAT, overriding the default JDK.  SO this should be safe to backport.

CLDR is supported since JDK11: https://www.oracle.com/java/technologies/javase/jdk11-suported-locales.html

### PR DESCRIPTION
**Pull request by akolarkunnu**
### Description
From JDK21 onwards a new warning has started to come, "WARNING: COMPAT locale provider will be removed in a future release". So, we have to avoid usage of COMPAT provider. We were setting exlpicitly to COMPAT locale provider in couple of places, this change is to convert COMPAT to CLDR locale provider. After this change, couple of tests started to fail becasue some locale data has minor changes in CLDR compared to COMPAT. For example, day and month short names of GERMAN "de" locale are different in CLDR and COMPAT, just need to add a . in the end for CLDR.

### Related Issues
Resolves #11550

Signed-off-by: Abdul Muneer Kolarkunnu <muneer.kolarkunnu@netapp.com>

### Check List
- [ ] ~New functionality includes testing.~
  - [x] All tests pass
- [ ] ~New functionality has been documented.~
  - [ ] ~New functionality has javadoc added~
- [ ] ~API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).~
- [ ] ~Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))~
- [x] Commits are signed per the DCO using --signoff
- [ ] ~Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))~
- [ ] ~Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Bump `org.apache.xmlbeans:xmlbeans` from 5.2.0 to 5.2.1 ([#13839](https://github.com/opensearch-project/OpenSearch/pull/13839))
 
 ### Changed
+- Changed locale provider from COMPAT to CLDR ([#11550](https://github.com/opensearch-project/OpenSearch/pull/13934))
```
[the remark this answers is not in the document]
Comment (akolarkunnu) [reply CHANGELOG.md]: corrected, thanks.

**Comment by reta** [comment]
@akolarkunnu could you please amend DCO check? it is failing now

**Comment by akolarkunnu** [comment]
Known flaky org.opensearch.indices.IndicesRequestCacheIT.testDeleteAndCreateSameIndexShardOnSameNode {p0={"opensearch.experimental.feature.pluggable.caching.enabled":"false"}} - #13792
Known flaky org.opensearch.indices.IndicesRequestCacheIT.testDeleteAndCreateSameIndexShardOnSameNode {p0={"search.concurrent_segment_search.enabled":"false"}} - #13600
Known flaky org.opensearch.indices.IndicesRequestCacheIT.testDeleteAndCreateSameIndexShardOnSameNode {p0={"search.concurrent_segment_search.enabled":"true"}} - #13600
Known flaky org.opensearch.indices.IndicesRequestCacheIT.testDeleteAndCreateSameIndexShardOnSameNode {p0={"opensearch.experimental.feature.pluggable.caching.enabled":"true"}} - #13792
Known flaky org.opensearch.snapshots.DeleteSnapshotIT.testDeleteSnapshot #9208

**Comment by akolarkunnu** [comment]
Can anybody please review this changes?

**Comment by reta** [comment]
@akolarkunnu could you please rebase your pull request against latest `main` (that will fix checks)? thank you

**Comment by akolarkunnu** [comment]
> @akolarkunnu could you please rebase your pull request against latest `main` (that will fix checks)? thank you

Closed this PR and created a fresh clean PR - https://github.com/opensearch-project/OpenSearch/pull/13988

**Pull request closed without merging**